src/services/user-group/hooks/index.js

```diff
@@ -76,20 +79,55 @@ const deleteWholeClassFromCourse = (hook) => {
 	});
 };
 
+const courseInviteHook = async hook => {
+	if (hook.path === 'courses' && hook.params.query && hook.params.query.link) {
+		const dbLink = await hook.app.service('link').get(hook.params.query.link) // link is used as "authorization"
+		delete hook.params.query.link
+		if (dbLink) return restrictToCurrentSchool(hook)
+	}
+
+	return restrictToUsersOwnCourses(hook)
+}
+
+const courseInvitePatchHook = async hook => {
+	const subsequentHooks = [globalHooks.hasPermission('USERGROUP_EDIT'), restrictToCurrentSchool, globalHooks.permitGroupOperation,
```

Comment:
More clarify is you create a hook for globalHooks.hasPermission('USERGROUP_EDIT') that is only try if it is want and all others must pass every time.

if ((hook.params.query || {} ).link){
    return globalHooks.hasPermission('USERGROUP_EDIT')
}
// or modified the hook link data
return context;